app/controllers/mechanics_controller.rb

```diff
@@ -0,0 +1,14 @@
+class MechanicsController < ApplicationController
+  def show
+    @mechanic = Mechanic.find(params[:id])
+  end
+
+  def edit
+    @mechanic = Mechanic.find(params[:id])
+    @ride = Ride.find(params[:ride_id])
+
+    @mechanic.rides << @ride
+
+    redirect_to mechanic_path(@mechanic)
```

Comment:
The content of this action is great! Just would be better put in a joins controller